### PR TITLE
Fix the Istio CNI doc per the rel-1.1 helm chart decoupling changes

### PR DESCRIPTION
**Pull request by tiswanso**
There is now a 2 step procedure to install Istio CNI and Istio separately, rather than the prior single/coupled helm install.

### CODE REVIEW

**Comment by tiswanso** [comment]
@john-a-joyce  Please review

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 1]
```diff
@@ -45,30 +45,73 @@ replaces the functionality provided by the `istio-init` container.
     1.  Refer to the [Hosted Kubernetes Usage](#hosted-kubernetes-usage) section for any non-default
         settings required.
 
-1.  Add the `--set istio_cni.enabled=true` setting to the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/) to include the installation of the Istio CNI plugin. For example:
+1.  Update Helm's local package cache per the instructions in the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/).
+
+    1.  If installing via `helm template` fetch the `istio-cni` template in addition to the other Istio
+        templates fetched per the [Istio Helm installation procedure](/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template)
+
+        {{< text bash >}}
```
Comment (john-a-joyce) [inline content/docs/setup/kubernetes/install/cni/index.md]: I think you need some kind of note here to indicate the CNI charts are pushed to the same locations that the istio charts are pushed to.   As we discussed directly I don't think you want to get in the version quagmire of which specific location to grab from.   For most people it may not be obvious that the CNI charts are pushed to the same locations.

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 2]
```diff
@@ -45,30 +45,73 @@ replaces the functionality provided by the `istio-init` container.
...
+    {{< /text >}}
+
+1.  Create and apply Istio CNI manifest.  Override the default configuration of the `istio-cni` Helm
+    chart's `logLevel` and `excludeNamespaces` parameters:
+
+    {{< text bash >}}
+    $ helm template $HOME/istio-fetch/istio-cni --name=istio-cni --namespace=istio-system \
     --set istio-cni.logLevel=info \
-    --set istio-cni.excludeNamespaces={"istio-system,foo_ns,bar_ns"} > $HOME/istio.yaml
+    --set istio-cni.excludeNamespaces={"istio-system,foo_ns,bar_ns"} > $HOME/istio-cni.yaml
```
Comment (john-a-joyce) [inline content/docs/setup/kubernetes/install/cni/index.md]: Do you want to keep the "istio-cni" part of "istio-cni.excludeNamespaces".  I am not sure if it breaks including it,  but it shouldn't be required.

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 3]
```diff
@@ -105,8 +148,8 @@ of many common Kubernetes environments.
     enable [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) in your cluster.
     *  Note for existing clusters this redeploys the nodes.
 
-1.  Install Istio via Helm including these options
-    `--set istio_cni.enabled=true --set istio-cni.cniBinDir=/home/kubernetes/bin`
+1.  Install Istio CNI via Helm including these options
```
Comment (john-a-joyce) [inline content/docs/setup/kubernetes/install/cni/index.md]: You lost the "istio_cni.enabled=true" part of the instructions.  Maybe reword to say "Install Istio CNI via Helm as described above, but also with this option."  Note also use of "these options" in original text is not proper.

File: content/docs/setup/kubernetes/install/cni/index.md (hunk 1 above)
Comment (tiswanso) [reply content/docs/setup/kubernetes/install/cni/index.md]: IMHO, since this is user facing doc they shouldn't care about those release & src repo details.  Doing the `helm repo add istio.io ...` instructions always makes these CNI instructions work.  So, I'm wary to add more complicated info here.

File: content/docs/setup/kubernetes/install/cni/index.md (hunk 2 above)
Comment (tiswanso) [reply content/docs/setup/kubernetes/install/cni/index.md]: you're right ... will fix

File: content/docs/setup/kubernetes/install/cni/index.md (hunk 3 above)
Comment (tiswanso) [reply content/docs/setup/kubernetes/install/cni/index.md]: I removed the `istio_cni.enabled` since the below refers to the istio-cni param.  Maybe I will add an example it like this:

```
$ helm install istio.io/istio-cni --name=istio-cni --namespace=istio-system --set cniBinDir=/home/kubernetes/bin
```

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 4]
```diff
@@ -45,30 +45,73 @@ replaces the functionality provided by the `istio-init` container.
     1.  Refer to the [Hosted Kubernetes Usage](#hosted-kubernetes-usage) section for any non-default
         settings required.
 
-1.  Add the `--set istio_cni.enabled=true` setting to the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/) to include the installation of the Istio CNI plugin. For example:
+1.  Update Helm's local package cache per the instructions in the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/).
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/install/cni/index.md]:
```suggestion
1.  Update Helm's local package cache with the instructions found in the [Istio Installation with Helm procedure](/docs/setup/kubernetes/install/helm/). 
    To install using the `helm template`, fetch the `istio-cni` template in addition
    to the other Istio templates you fetched when following [option 1 of the Istio Helm installation procedure](/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template).

```

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 5]
```diff
@@ -45,30 +45,73 @@ replaces the functionality provided by the `istio-init` container.
...
+
+    1.  If installing via `helm template` fetch the `istio-cni` template in addition to the other Istio
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/install/cni/index.md]: To help simplify things for readers, I suggest you add this content to the main step. I've left a suggestion that does this for you to be able to quickly implement it. I'm also removing the lines that then become redundant.
```suggestion
```

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 6]
```diff
@@ -45,30 +45,73 @@ replaces the functionality provided by the `istio-init` container.
...
+        templates fetched per the [Istio Helm installation procedure](/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template)
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/install/cni/index.md]:
```suggestion
```

File: content/docs/setup/kubernetes/install/cni/index.md (hunk 1 above)
Comment (rcaballeromx) [reply content/docs/setup/kubernetes/install/cni/index.md]: @john-a-joyce We can add the details of the implementation in a separate document at a later date. Adding the details here could confuse and over-complicate the procedure.

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 7]
```diff
@@ -45,30 +45,73 @@ replaces the functionality provided by the `istio-init` container.
...
 
     {{< text bash >}}
-    $ helm template install/kubernetes/helm/istio --name istio --namespace istio-system \
-      --set istio_cni.enabled=true > $HOME/istio.yaml
+    $ kubectl create namespace istio-system
     {{< /text >}}
 
+1.  Install Istio CNI.
+
+    1.  Option 1.  Render and apply the Istio CNI Helm template.
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/install/cni/index.md]: Do not use numbered lists for alternative options. The numbers tell readers that they should run one command after the other and that is not the case here. 

```suggestion
    - Option 1.  Render and apply the Istio CNI Helm template.
```

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 8]
```diff
@@ -45,30 +45,73 @@ replaces the functionality provided by the `istio-init` container.
...
+
+        {{< text bash >}}
+        $ helm template $HOME/istio-fetch/istio-cni --name=istio-cni --namespace=istio-system | kubectl apply -f -
+        {{< /text >}}
+
+    1.  Option 2.  Install via Helm and Tiller via `helm install`.
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/install/cni/index.md]:
```suggestion
    - Option 2.  Install via Helm and Tiller via `helm install`.
```

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 9]
```diff
@@ -105,8 +148,14 @@ of many common Kubernetes environments.
     enable [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) in your cluster.
     *  Note for existing clusters this redeploys the nodes.
 
-1.  Install Istio via Helm including these options
-    `--set istio_cni.enabled=true --set istio-cni.cniBinDir=/home/kubernetes/bin`
+1.  Install Istio CNI via Helm including this option
+    `--set cniBinDir=/home/kubernetes/bin`
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/install/cni/index.md]:
```suggestion
    `--set cniBinDir=/home/kubernetes/bin`. For example, the following `helm install`
    command sets the `cniBinDir` value for GKE setups:
```

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 10]
```diff
@@ -105,8 +148,14 @@ of many common Kubernetes environments.
...
+
+    1.  For example, the following `helm install` command sets the `cniBinDir` value for GKE setups:
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/install/cni/index.md]:
```suggestion
```

File: content/docs/setup/kubernetes/install/cni/index.md [hunk 11]
```diff
@@ -105,8 +149,14 @@ of many common Kubernetes environments.
     enable [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) in your cluster.
     *  Note for existing clusters this redeploys the nodes.
 
-1.  Install Istio via Helm including these options
-    `--set istio_cni.enabled=true --set istio-cni.cniBinDir=/home/kubernetes/bin`
+1.  Install Istio CNI via Helm including this option
+    `--set cniBinDir=/home/kubernetes/bin`. For example, the following `helm install`
+    command sets the `cniBinDir` value for GKE setups:
+
+
```
Comment (tiswanso) [inline content/docs/setup/kubernetes/install/cni/index.md]: need to fix up the spacing and newlines here... it rendered funny

**Comment by tiswanso** [comment]
@rcaballeromx -- I took all your suggestions basically as is and rebased and fix lint issues.  Should be good to go, IMO.
